qa-tests-backend/src/test/groovy/NetworkBaselineTest.groovy

```diff
@@ -373,4 +427,130 @@ class NetworkBaselineTest extends BaseSpecification {
+            }
+            return externalBaseline
+        }
+
+        assert externalBaseline
+
+        def peerStatus1 = externalBaseline.getBaselineList().find { it.getPeer().getEntity().getName() == EXTERNAL_IP1 }
+        def peer1 = peerStatus1.getPeer()
+
+        assert peer1
+        def expectedEntity1 = peer1.getEntity()
+        verifyAll(expectedEntity1) {
+            type == NetworkFlowOuterClass.NetworkEntityInfo.Type.EXTERNAL_SOURCE
+            name == EXTERNAL_IP1
+            discovered == true
+        }
+
+        assert peer1.getPort() == 53
+        assert peer1.getProtocol() == NetworkFlowOuterClass.L4Protocol.L4_PROTOCOL_TCP
+
+        def peerStatus2 = externalBaseline.getBaselineList().find { it.getPeer().getEntity().getName() == EXTERNAL_IP2 }
+        def peer2 = peerStatus2.getPeer()
+
+        assert peer2
+        def expectedEntity2 = peer2.getEntity()
+        verifyAll(expectedEntity2) {
+            type == NetworkFlowOuterClass.NetworkEntityInfo.Type.EXTERNAL_SOURCE
+            name == EXTERNAL_IP2
+            discovered == true
+        }
+
+        assert peer2.getPort() == 53
+        assert peer2.getProtocol() == NetworkFlowOuterClass.L4Protocol.L4_PROTOCOL_TCP
+
+        def peerStatus3 = externalBaseline.getBaselineList().find { it.getPeer().getEntity().getName() == EXTERNAL_IP3 }
+        def peer3 = peerStatus3.getPeer()
+
+        assert peer3
+        def expectedEntity3 = peer3.getEntity()
+        verifyAll(expectedEntity3) {
+            type == NetworkFlowOuterClass.NetworkEntityInfo.Type.EXTERNAL_SOURCE
+            name == EXTERNAL_IP3
+            discovered == true
+        }
+
+        assert peer3.getPort() == 80
+        assert peer3.getProtocol() == NetworkFlowOuterClass.L4Protocol.L4_PROTOCOL_TCP
+
+        assert externalBaseline.getTotalBaseline() == 3
+        assert externalBaseline.getTotalAnomalous() == 0
+
+        def status = NetworkBaselineServiceOuterClass.NetworkBaselinePeerStatus.Status.ANOMALOUS
+        def modifiedPeer = modifyBaseline(peer1, status)
+
+        NetworkBaselineService.modifyBaselineStatusForPeers(deploymentUid, modifiedPeer)
+
+        def externalBaselineAfter = evaluateWithRetry(30, 4) {
+            def externalBaselineAfter = NetworkBaselineService.getNetworkBaselineForExternalFlows(deploymentUid)
+            if (externalBaselineAfter.totalAnomalous + externalBaselineAfter.totalBaseline == 0) {
+                throw new RuntimeException(
+                    "No peers in baseline for deployment ${deploymentUid} yet. Baseline is ${externalBaselineAfter}"
+                )
+            }
+            return externalBaselineAfter
+        }
+
+        assert externalBaselineAfter.getTotalBaseline() == 1
+        assert externalBaselineAfter.getTotalAnomalous() == 2
+    }
+
+    def modifyBaseline(NetworkBaselineServiceOuterClass.NetworkBaselineStatusPeer peer, NetworkBaselineServiceOuterClass.NetworkBaselinePeerStatus.Status status) {
```

Comment:
The name is inaccurate as this does not actually modify the baseline, but just creates an object of type `NetworkBaselinePeerStatus` which can be used to modify the baseline.